app/javascript/packages/document-capture/components/barcode-attention-warning.tsx

```diff
@@ -25,15 +26,23 @@ function BarcodeAttentionWarning({ onDismiss, pii }: BarcodeAttentionWarningProp
   const { formData } = useContext(UploadContext);
 
   async function skipAttention() {
-    await window.fetch(DOCUMENT_CAPTURE_ERRORS_API_URL, {
-      method: 'DELETE',
-      body: toFormData({ document_capture_session_uuid: formData.document_capture_session_uuid }),
-    });
+    await Promise.all([
+      trackEvent('IdV: barcode warning continue clicked'),
+      window.fetch(DOCUMENT_CAPTURE_ERRORS_API_URL, {
+        method: 'DELETE',
+        body: toFormData({ document_capture_session_uuid: formData.document_capture_session_uuid }),
+      }),
+    ]);
     window.onbeforeunload = null;
     const form = document.querySelector<HTMLFormElement>('.js-document-capture-form');
     form?.submit();
   }
 
+  function handleDismiss() {
+    trackEvent('IdV: barcode warning retake photos clicked');
+    onDismiss();
+  }
```

Comment:
Seeing this gave me an idea, would it be too weird to make a version of trackEvent that chains? Or calls a callback when finished?

```suggestion
  function handleDismiss() {
    trackEvent('IdV: barcode warning retake photos clicked').then(onDismiss);
  }
```

But it looks like `trackEvent` is async, so that would make things take longer if it actually waited, and misleading if it didn't wait

Reply:
Yeah, `trackEvent` is asynchronous, and as implemented here it's treated as a fire-and-forget, with the expectation that they'll not leaving the page so there's a good chance the event will finish logging successfully. I'd be hesitant to `await` the result because of the UX implication you mentioned. Typically in those cases I would use a `SpinnerButton`, similar to what we're doing for the "Continue" button.

As far as code style goes, historically I've liked using some of the Lodash utility functions for this kind of thing ([`over`](https://lodash.com/docs/4.17.15#over)?), so you could do something like:

```suggestion
  const trackClickEvent = () => trackEvent('IdV: barcode warning retake photos clicked');
  const onRetakePhotosClick = over([trackClickEvent, onDismiss]);
```